src/expint.jl

```diff
@@ -350,6 +348,11 @@ function En_imagbranchcut(ν::Number, z::Number)
     return -2 * lgammasign * e1 * π * e2 * exp((ν-1)*log(complex(a)) - lgamma) * im
 end
 
+function En_safeexpmult(z, a)
+    zexp = exp(z)
+    return isinf(zexp) ? exp(z + log(a)) : zexp*a
```

Comment:
You also need to check for underflow here.  For example, `100 * exp(-746.0)` gives `0.0` due to underflow, but `exp(-746 + log(100))` gives `1.04e-322`.

Do you also need to worry about `a < 0`?   That is, shouldn't you do `sign(a) * exp(z + log(abs(a)))`, or does that case never come up where this function is called?